### PR TITLE
License identifier update

### PR DESCRIPTION
**Pull request by mkleinschnitker**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

- checked header of src files for each formula
- new identifiers based on https://spdx.org/licenses/

### CODE REVIEW

File: Formula/beagle.rb [hunk 1]
```diff
@@ -3,7 +3,7 @@ class Beagle < Formula
   homepage "https://github.com/beagle-dev/beagle-lib"
   url "https://github.com/beagle-dev/beagle-lib/archive/v3.1.2.tar.gz"
   sha256 "dd872b484a3a9f0bce369465e60ccf4e4c0cd7bd5ce41499415366019f236275"
-  license "LGPL-3.0"
+  license "LGPL-3.0-or-later"
```
Comment (kthchew) [inline Formula/beagle.rb:6]: Some files in the project appear to be licensed under `GPL-3.0-or-later`, such as https://github.com/beagle-dev/beagle-lib/blob/master/m4/ax_gcc_x86_cpuid.m4

Others are licensed under `LGPL-3.0-or-later`, such as https://github.com/beagle-dev/beagle-lib/blob/master/libhmsbeagle/BeagleImpl.h

The entire project seems to fall under LGPL 3.0. I think putting both GPL 3 and later and LGPL 3 and later is appropriate here.

File: Formula/check.rb [hunk 2]
```diff
@@ -3,7 +3,7 @@ class Check < Formula
   homepage "https://libcheck.github.io/check/"
   url "https://github.com/libcheck/check/releases/download/0.15.2/check-0.15.2.tar.gz"
   sha256 "a8de4e0bacfb4d76dd1c618ded263523b53b85d92a146d8835eb1a52932fa20a"
-  license "LGPL-2.1"
+  license "LGPL-2.1-or-later"
```
Comment (kthchew) [inline Formula/check.rb:6]: A few files are licensed under GPL 3.0 or later, like https://github.com/libcheck/check/blob/master/m4/ax_create_stdint_h.m4

A few other older files are licensed under GPL 2.0 or later, like https://github.com/libcheck/check/blob/master/m4/ax_cflags_warn_all_ansi.m4

The project overall seems to be licensed under LGPL 2.1 or later.

**Review by kthchew — COMMENTED** [review]
Most of these look good from a quick check, though I just have 2 suggestions.

File: Formula/beagle.rb (hunk 1 above)
Comment (mkleinschnitker) [reply Formula/beagle.rb:6]: If more than one licence identifier is given, are they separated by commas? 

Something like that: `license: "LGPL-3.0-or-later, GPL-3.0-or-later"`

File: Formula/check.rb (hunk 2 above)
Comment (mkleinschnitker) [reply Formula/check.rb:6]: Is it appropriate to use all three license identifier (GPL 3.0 or later, GPL 2.0 or later and  LGPL-2.1-or-later) or just LGPL-2.1-or-later?

File: Formula/beagle.rb (hunk 1 above)
Comment (kthchew) [reply Formula/beagle.rb:6]: Check out the documentation about `all_of:` here: https://docs.brew.sh/License-Guidelines#complex-spdx-license-expressions

`all_of` would likely be more appropriate than `any_of` here. So something like:

```
license all_of: ["LGPL-3.0-or-later", "GPL-3.0-or-later"]
```

File: Formula/check.rb (hunk 2 above)
Comment (kthchew) [reply Formula/check.rb:6]: It's probably safest to list both GPL 2.0 or later, as well as LGPL 2.1 or later (since GPL 3.0 falls under GPL 2.0 or later).

**Comment by mkleinschnitker** [comment]
Hi @kthchew,

I updated the license identifiers of the two formulas.

**Pull request closed without merging**